starlark/testdata/time.star

```diff
@@ -0,0 +1,119 @@
+
+assert.eq(t - t, time.parse_duration("0s"))
+
+d = time.parse_duration("1s")
+assert.eq(d - d, time.parse_duration("0"))
+assert.eq(d + d, time.parse_duration("2s"))
+assert.eq(d * 5, time.parse_duration("5s"))
+assert.eq(time.parse_duration("0s") + time.parse_duration("3m35s"), time.parse_duration("3m35s"))
+
+d2 = time.parse_duration("10h")
+# duration attributes
+assert.eq(10.0, d2.hours)
+assert.eq(10*60.0, d2.minutes)
+assert.eq(10*60*60.0, d2.seconds)
+assert.eq(10*60*60*1000, d2.milliseconds)
+assert.eq(10*60*60*1000000, d2.microseconds)
+assert.eq(10*60*60*1000000000, d2.nanoseconds)
+
+# duration == duration = boolean
+# duration != duration = boolean
+assert.eq(time.parse_duration("1h"), time.parse_duration("1h"))
+assert.ne(time.parse_duration("1h"), time.parse_duration("1m"))
+# duration < duration = boolean
+assert.lt(time.parse_duration("1m"), time.parse_duration("1h"))
+assert.true(not time.parse_duration("1h") < time.parse_duration("1h"))
+assert.true(not time.parse_duration("1h") < time.parse_duration("1m"))
+# duration <= duration = boolean
+assert.true(time.parse_duration("1m") <= time.parse_duration("1h"))
+assert.true(time.parse_duration("1h") <= time.parse_duration("1h"))
+assert.true(not time.parse_duration("1h") <= time.parse_duration("1m"))
+# duration > duration = boolean
+assert.true(not time.parse_duration("1m") > time.parse_duration("1h"))
+assert.true(not time.parse_duration("1h") > time.parse_duration("1h"))
+assert.true(time.parse_duration("1h") > time.parse_duration("1m"))
+# duration >= duration = boolean
+assert.true(not time.parse_duration("1m") >= time.parse_duration("1h"))
+assert.true(time.parse_duration("1h") >= time.parse_duration("1h"))
+assert.true(time.parse_duration("1h") >= time.parse_duration("1m"))
+
+# duration + duration = duration
+assert.eq(d2 + d, time.parse_duration("10h01s"))
+# duration + time = time
+assert.eq(d2 + time.parse_time("2011-04-22T13:33:48Z"), time.parse_time("2011-04-22T23:33:48Z"))
+# duration - duration = duration
+assert.eq(d2 - d, time.parse_duration("9h59m59s"))
+# duration / duration = float
+assert.eq(d2 / time.parse_duration("16m"), 37.5)
+assert.fails(lambda: d2 / time.parse_duration("0"), "division by zero")
+# duration / int = duration
+assert.eq(d2 / 20, time.parse_duration("30m"))
+assert.fails(lambda: d2 / 0, "division by zero")
+# duration // duration = int
+assert.eq(d2 // time.parse_duration("16m"), 37)
+assert.fails(lambda: d2 // time.parse_duration("0"), "division by zero")
+# duration * int = duration
+assert.eq(d * 1000, time.parse_duration("16m40s"))
+assert.fails(lambda: d2 // 0, "division by zero")
+
+
+before = time.now()
+time.sleep(10 * time.millisecond)
+assert.true((time.now() - before).nanoseconds >= time.parse_duration("10ms").nanoseconds)
+
+# time(year=..., month=..., day=..., hour=..., minute=..., second=..., nanosecond=..., location=...)
+t1 = time.time(2009, 6, 12, 12, 6, 10, 99, "US/Eastern")
```

Comment:
I suggest we get rid of the positional argument constructor: it seems like a recipe for mistakes, whereas `year=2009, month=6` etc is self-documenting.

Reply:
(I realize I said the opposite for the other functions, but they all have 1 parameter with a nonce name like "x".)